src/topology/algebra/order/lower_topology.lean

```diff
@@ -0,0 +1,150 @@
+basis.
+
+## Implementation notes
+
+Approach inspired by `order_topology` from topology.algebra.order.basic
+
+## References
+
+* [Gierz et al, A Compendium of Continuous Lattices][GierzEtAl1980]
+
+## Tags
+
+lower topology, preorder
+-/
+
+universe u
+variable {α : Type u}
+
+open set topological_space
+
+/--
+Type synonym for a preorder equipped with the lower topology
+-/
+@[derive preorder, nolint unused_arguments]
+def lower (α : Type u) [preorder α] := α
+
+instance (α : Type u) [preorder α] [p : nonempty α] : nonempty (lower α) := p
+
+instance (α : Type u) [preorder α] [p : inhabited α] : inhabited (lower α) := p
+
+/--
+The lower topology is the topology generated by the complements of the closed intervals to infinity
+-/
+def lower_topology (α : Type u) [preorder α] : topological_space α :=
+generate_from {s | ∃ a, (Ici a)ᶜ = s}
+
+instance lower.topological_space (α : Type u) [preorder α] :
+  topological_space (lower α) := lower_topology (lower α)
+
+namespace lower_topology
+section pre_order
+
+
+variable [preorder α]
+
+lemma is_open_iff_generate_Ici_comp {s : set (lower α)} :
+  is_open s ↔ generate_open {s | ∃ a, (Ici a)ᶜ = s} s := iff.rfl
+
+/-
+Left-closed right-infinite intervals [a,∞) are closed in the lower topology
+-/
+lemma is_closed_Ici (a : lower α) : is_closed (Ici a) :=
+is_open_compl_iff.1 $ generate_open.basic _ ⟨a, rfl⟩
+
+/-
+The upper closure of a finite subset is closed in the lower topology
+-/
+lemma is_closed_upper_closure (F : set (lower α)) (h : F.finite) :
+  is_closed (upper_closure F : set (lower α)) :=
+begin
+  simp only [← upper_set.infi_Ici, upper_set.coe_infi],
+  exact is_closed_bUnion h (λ a h₁, is_closed_Ici a),
+end
+
+/-
+Every subset open in the lower topology is a lower set
+-/
+lemma is_lower_set_of_is_open {s : set (lower α)} (h : is_open s) : is_lower_set s :=
+begin
+  rw is_open_iff_generate_Ici_comp at h,
+  induction h,
+  case generate_open.basic : u h { obtain ⟨a, rfl⟩ := h, exact (is_upper_set_Ici a).compl },
+  case univ : { exact is_lower_set_univ },
+  case inter : u v hu1 hv1 hu2 hv2 { exact hu2.inter hv2 },
+  case sUnion : _ _ ih { exact is_lower_set_sUnion ih },
+end
+
+lemma lower_closed_is_upper {s : set (lower α)} (h : is_closed s) : is_upper_set s :=
+is_lower_set_compl.1 $ is_lower_set_of_is_open h.is_open_compl
+
+/-
+The closure of a singleton {a} in the lower topology is the left-closed right-infinite interval
+[a,∞)
+-/
+lemma closure_singleton (a : lower α) : closure {a} = Ici a :=
+subset_antisymm (closure_minimal (λ b h, h.ge) $ is_closed_Ici a) $
+  is_upper_set.Ici_subset (lower_closed_is_upper is_closed_closure) (subset_closure rfl)
+
+/--
+The non-empty complements of the upper closures of finite subsets are a collection of lower sets
+which form a basis for the lower topology
+-/
+def lower_basis (α : Type u) [preorder α] :=
+{s : set α | ∃ (F : set α), F.finite ∧
+  ↑(upper_closure F).compl = s }
```

Comment:
Yes it's better to keep `α` explicit here. You may achieve that with `variable (α)` but here repeating `(α : Type u) [preorder α]` isn't too long :)
```suggestion
{s : set α | ∃ (F : set α), F.finite ∧ ↑(upper_closure F).compl = s}
```
looks nicer.